x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -382,15 +396,16 @@ private void openJob(JobTask jobTask) {
                                 if (unwrapped instanceof DocumentMissingException || unwrapped instanceof ResourceNotFoundException) {
                                     jobTask.markAsCompleted();
                                 } else if (autodetectProcessManager.isNodeDying() == false) {
-                                    jobTask.markAsFailed(e);
```

Comment:
Possibly this is one where we really should fail the task rather than the job, and let the job go into the `closed` state.  Failure to finalize means that the job was intended to be closing, and we just failed to update some fields on the job config like finish time.  The fact that somebody had deliberately asked to close the job makes me think that in this case it's better to let it close even though the finish time didn't get set.

Reply:
I also contemplated this one. The reason I decided to also `fail` the job in this case was so that if something goes wrong, it is clear that ... something went wrong. But I guess if updating a field failed we should at least have something in the logs.